### PR TITLE
[move-vm] support bytecode for vector ops in the Move VM

### PR DESCRIPTION
**Pull request by mengxu-fb**
The interpretation logic for vector-related bytecodes are added in the
Move VM, including the gas charging logic. Would love to receive some
feedback from @vgao1996  @tzakian  on the gas metering.

This PR does not remove the native implementation for the `Vector`
module in move-stdlib yet, so both versions co-exist for now.

## Motivation

More progress on the vector-related bytecode implementation

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

New test cases are added. And since we only have IR compiler support here, these tests are added in the ~~ir-testsuite~~ `move-vm/tests` under the new transactional testing infra.

### CODE REVIEW

**Review by tnowacki — COMMENTED** [review]
Things look mostly good to me. Unfortunately I don't have time to do a deeper review before being away for a bit. I am mostly concerned about test coverage. If that looks good, these changes look great and I'd be fine with someone else stamping/approving them :)

File: language/move-vm/transactional-tests/tests/builtins/vector_ops_destroy_non_empty.mvir [hunk 1]
```diff
@@ -0,0 +1,10 @@
+//# run
```
Comment (tnowacki) [inline language/move-vm/transactional-tests/tests/builtins/vector_ops_destroy_non_empty.mvir:1]: Maybe have another script in this file that succeeds? Maybe even one with a type without drop if you want to get fancy :)

**Review by tnowacki — COMMENTED** [review]
Looks like we are missing tests for pack/unpack of non-zero sizes? Or am I missing something 

And for the test categorization? Whats the current layout. One test for each error code then a big test that uses everything? Might make more sense to have valid/invalid tests for each op instead? I'm not really sure

**Comment by mengxu-fb** [comment]
The recent push include two commits
- renamed `vec_instantiations` in the loader data struct to `single_signature_token_map`, as the old wording is too narrow. This mapping is useful beyond vector bytecode, especially with the possibility of adding bytecode support for set and map data structures.
- added many more test cases for the vector bytecode. Now each vector bytecode that can error, i.e., `VecPopBack`, `VecImmBorrow`, `VecMutBorrow`, `VecSwap`, and `VecUnpack` now have both positive and negative tests. Those don't error (`VecLen`, `VecPushBack`, `VecPack`) have positive tests. There are also a few extended tests that covers the operations of several bytecode combined.

File: language/move-vm/transactional-tests/tests/builtins/vector_ops_destroy_non_empty.mvir (hunk 1 above)
Comment (mengxu-fb) [reply language/move-vm/transactional-tests/tests/builtins/vector_ops_destroy_non_empty.mvir]: The one without `drop` ability should be caught by the bytecode verifier I guess? As tested in this test case: https://github.com/diem/diem/blob/main/language/bytecode-verifier/transactional-tests/tests/locals_safety/vector_ops_non_droppable_resource_not_destroyed.mvir

**Review by tnowacki — APPROVED** [review]
